client/src/templates/Challenges/components/scene/scene.tsx

```diff
@@ -131,9 +131,40 @@ export function Scene({
     }
   }
 
+  const buildTranscript = () => {
+    let transcript = '';
+    commands.forEach(command => {
+      console.log(command);
+      if (
+        command.character != null &&
+        command.dialogue != null &&
+        command.startTime != null
+      ) {
+        transcript =
+          transcript +
+          '\n' +
+          command.startTime +
+          '&nbsp;&nbsp;&nbsp;&nbsp;' +
+          command.character +
+          ':' +
+          ' ' +
+          command.dialogue.text;
+      }
+    });
+    return transcript;
+  };
+
+  const toggleTranscript = () => {
+    const transcriptElement = document.getElementById('transcript')!;
+    if (transcriptElement.innerText == '') {
+      transcriptElement.innerHTML = buildTranscript();
```

Comment:
innerHTML feels like an anti-pattern in react. Can we not use a `useState` and conditionally render the transcript? We're not even using HTML elements in the buildTranscript function.

Reply:
I originally had `innerText` but was experimenting with adding tabs and forgot to set it back.